### PR TITLE
DSDEGP-1768 - Verify if User is registered to FC at login

### PR DESCRIPTION
**Pull request by LeoForconesi**
### Description

_This code checks if the user is registered to Firecloud site at the login, providing some messages describing possible solutions. Eg. to register to firecloud._
_The motivation for this code is to provide an extra security level to users that uses this app and provide recovery options from possible system errors or user mistakes at the login_
_Jira's TIcket link: https://broadinstitute.atlassian.net/browse/DSDEGP-1768_

----

### Checklist

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=189431877).
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

**Comment by pshapiro4broad** [comment]
One problem I can see while testing this is that, in the case where the cookie is used to log in, the Sign In page isn't locked out, and no UI feedback is presented that the user should wait. So if the user clicks Sign In, sees the google dialog come up and go away quickly, and is returned to the Sign In page, they can click the button again. Instead we want a UI flow where they're locked out of this. One way is to eliminate our Sign In page, e.g. the user is always sent to the google login dialog right away. We would have to clear the google cookies on Sign Out, so it has an effect.

File: src/app/file-manager/services/login.service.ts [hunk 1]
```diff
@@ -16,34 +18,52 @@ const googleConfig = {
...
+      }
+      );
     } else {
       alert('Implement web login');
       return null;
     }
   }
 
+  setGoogleOptions(): Observable<any> {
+    return new Observable<any>(
```
Comment (pshapiro4broad) [inline src/app/file-manager/services/login.service.ts]: Do you need to make a new observable each time this method is called? Or could we make this object once and return the same one each time?

Comment (LeoForconesi) [reply src/app/file-manager/services/login.service.ts]: This observable has been removed and replaced by a constructor call to configure google options.

**Comment by LeoForconesi** [comment]
You're right, this a flaw we need to correct.
Regarding google's dialog, instead of cleaning the app's cache it would be preferable to perform a proper google's account logout. For this we got an existing Jira's story that we could implement on the upcoming sprint if its ok for you. 

On the other hand, to disable or block electron's parent window when google's pop-up is displayed we need to investigate how to accomplish that. Maybe we could add this aspect as an acceptance criteria in google's logout story. https://broadinstitute.atlassian.net/browse/DSDEGP-1776
I'll wait for your comments on how to proceed with this.

**Comment by LeoForconesi** [comment]
Update. 
- We managed a workaround to lock Sign in page.
- We would need to define and implement the need of a Ui feedback such as a load spinner for loading moments.

**Pull request merged**